pkg/build/controller/factory/factory.go

```diff
@@ -112,7 +112,7 @@ func (factory *BuildControllerFactory) CreateDeleteController() controller.Runna
 		RetryManager: controller.NewQueueRetryManager(
 			queue,
 			cache.MetaNamespaceKeyFunc,
-			limitedLogAndRetry(factory.BuildUpdater, 30*time.Minute),
+			controller.RetryNever,
```

Comment:
the sync loop will ultimately retry this whole thing if the state of "build pod with missing build" continues:
https://github.com/openshift/origin/blob/master/pkg/build/controller/factory/factory.go#L353